ksqldb-common/src/main/java/io/confluent/ksql/function/GenericsUtil.java

```diff
@@ -291,4 +324,35 @@ public static boolean instanceOf(final ParamType schema, final SqlArgument insta
 
     return true;
   }
+
+  public static void reserveGenerics(
+      final ParamType schema,
+      final SqlArgument argument,
+      final Map<GenericType, SqlType> reservedGenerics
+  ) {
+    if (!GenericsUtil.instanceOf(schema, argument)) {
```

Comment:
re: above: I think even if we want the later error handling in this method this scenario should be covered in the call to `resolveGenerics`